identity/aziot-identityd/src/identity.rs

```diff
@@ -1002,6 +996,45 @@ impl IdentityManager {
     }
 }
 
+fn get_prev_modules(
+    prev_settings_path: &std::path::Path,
+    prev_device_info_path: &std::path::Path,
+    curr_hub_device_info: HubDeviceInfo,
+) -> std::collections::BTreeSet<aziot_identity_common::ModuleId> {
+    if !prev_settings_path.exists() || !prev_device_info_path.exists() {
+        return std::collections::BTreeSet::default();
+    }
+
+    let prev_hub_device_info = match HubDeviceInfo::new(prev_device_info_path) {
+        Ok(device_info) => device_info,
+        Err(err) => {
+            log::warn!("Ignoring invalid device info backup: {}", err);
+
+            return std::collections::BTreeSet::default();
+        }
+    };
+
+    // Only consider the previous Hub modules if the current and previous Hub devices match.
+    if prev_hub_device_info != Some(curr_hub_device_info) {
+        return std::collections::BTreeSet::default();
```

Comment:
Should we clear the prev hub device info?

Reply:
It's overwritten by the caller of this function (line 986) so no need to clear it here.